tests/5.0/target/test_target_device.c

```diff
@@ -13,28 +13,33 @@
 #include <omp.h>
 #include "ompvv.h"
 
-#define N 1028
+#define N 1024
 
 // Required for 'device(ancestor: 1)
 #pragma omp requires reverse_offload
 
 int test_target_device_ancestor() {
 
-    int i, which_device;
+    int which_device;
     int a[N];
     int errors = 0; 
+    int is_shared_env = 0;
+    which_device = 0;
 
     for (int i = 0; i < N; i++) {
         a[i] = i;
     }
 
     OMPVV_TEST_AND_SET(errors, omp_get_num_devices() <= 0);
-    OMPVV_ERROR_IF(omp_get_num_devices() <= 0, "Since no target devices were found, this test"
+    OMPVV_WARNING_IF(omp_get_num_devices() <= 0, "[SKIPPED] Since no target devices were found, this test"
                                                  "will be skipped.");
+    OMPVV_TEST_AND_SET_SHARED_ENVIRONMENT(is_shared_env);
+    OMPVV_WARNING_IF(is_shared_env != 0, "[WARNING] target_device_ancestor() test may not be able to detect errors if the target system supports shared memory.")
 
-    if (omp_get_num_devices() > 0) {
+    #pragma omp target //Run on the default device, which is the host for device_num = 0
```

Comment:
Same issue with the implicit `firstprivate(which_device)` as above.

Reply:
Updated.

Reply:
Updated the C version to have the nested target region as the Fortran version does.